### PR TITLE
Docs: Add outline and basic content for Tech Preview docs

### PR DESCRIPTION
**Pull request by BeenzSyed**
Outline for NIA/Consul-Terraform-Sync Tech Preview docs with some base content.

### CODE REVIEW

File: website/pages/docs/nia/index.mdx [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+---
+
+# Network Infrastructure Automation
+
+Network Infrastructure Automation (NIA) enables dynamic updates to network infrastructure devices triggered by service changes. Consul-Terraform-Sync utilizes Consul as a data source that contains networking information about services and monitors those services. Terraform is used as the underlying automation tool and leverages the Terraform provider ecosystem to drive relevant change to the network infrastructure.
+
+Consul-Terraform-Sync executes one or more automation tasks with the most recent service variable values from the Consul service catalog. Each task consists of a runbook automation written as a Consul-Terraform-Sync compatible Terraform module using resources and data sources for the underlying network infrastructure. The “Consul-Terraform-Sync” daemon runs on the same node as a Consul agent.
```
Comment (lornasong) [inline website/pages/docs/nia/index.mdx:13]: It seems like we're using lower-case `consul-terraform-sync` to refer to the daemon and upper-case to refer to the product? if so, maybe "“Consul-Terraform-Sync” daemon" => "consul-terraform-sync daemon"

File: website/pages/docs/nia/installation/configuration.mdx [hunk 2]
```diff
@@ -0,0 +1,11 @@
+---
+layout: docs
+page_title: Configure Consul-Terraform-Sync
+sidebar_title: Configuration
+description: >-
+  Consul-Terraform-Sync requires a Terraform Provider, a Terraform Module and a running Consul Cluster outside of the consul-tf-sync daemon.
```
Comment (lornasong) [inline website/pages/docs/nia/installation/configuration.mdx:6]:
```suggestion
  Consul-Terraform-Sync requires a Terraform Provider, a Terraform Module and a running Consul Cluster outside of the consul-terraform-sync daemon.
```

**Review by lornasong — COMMENTED** [review]
Thanks for writing up this outline and all of this content 🎉 . I left 2 minor nits around the sync name - feel free to address as you see best. I'll take a look again when I see the netlify preview!

File: website/pages/docs/nia/index.mdx (hunk 1 above)
Comment (BeenzSyed) [reply website/pages/docs/nia/index.mdx:13]: I was using it interchangeably. That's a good idea, we can use all lower case for the daemon and the first letters upper case for the product. Thank you!

Comment (eikenb) [reply website/pages/docs/nia/index.mdx:13]: Not a big issue, but just a bit of CLI/Unix nomenclature. It is still usually referred to as a binary, program, executable, etc. The term daemon is more used to reference how it runs in the background.

File: website/pages/docs/nia/installation/install.mdx [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+layout: docs
+page_title: Install Consul and Consul-Terraform-Sync
+sidebar_title: Installation
+description: >-
+  Consul-Terraform-Sync is a daemon that runs alongside Consul. Consul-Terraform-Sync is not included with the Consul binary and will need to be installed separately.
+---
+
+# Installing Consul
+
+Consul-Terraform-Sync is a daemon that runs alongside Consul, similar to other Consul ecosystem tools like Consul Template. Consul-Terraform-Sync is not included with the Consul binary and will need to be installed separately. First, we recommend installing Consul. For simplicity, we are using a local Consul agent in development mode which can be started with consul agent -dev. To quickly set up a local Consul agent, refer to the getting started [guide](https://learn.hashicorp.com/tutorials/consul/get-started-install). The Consul agent must be running to dynamically update network devices.
```
Comment (lornasong) [inline website/pages/docs/nia/installation/install.mdx:11]:
```suggestion
Consul-Terraform-Sync is a daemon that runs alongside Consul, similar to other Consul ecosystem tools like Consul Template. Consul-Terraform-Sync is not included with the Consul binary and will need to be installed separately. First, we recommend installing Consul. For simplicity, we are using a local Consul agent in development mode which can be started with `consul agent -dev`. To quickly set up a local Consul agent, refer to the getting started [guide](https://learn.hashicorp.com/tutorials/consul/get-started-install). The Consul agent must be running to dynamically update network devices.
```

Comment (lornasong) [reply website/pages/docs/nia/installation/install.mdx:11]: suggesting some formatting around the cli command to separate it out from the other test

**Review by lornasong — APPROVED** [review]
this looks good to me!

File: website/pages/docs/nia/index.mdx [hunk 4]
```diff
@@ -0,0 +1,33 @@
+---
+layout: docs
+page_title: Network Infrastructure Automation <sup>Tech Preview</sup>
+sidebar_title: Network Infrastructure Automation <sup>Tech Preview</sup>
+description: >-
+  Network Infrastructure Automation (NIA) relies on a declarative, workflow and service driven network automation architecture. Consul-Terraform-Sync is a daemon based architecture that utilizes Terraform as the underlying automation tool and leverages the Terraform provider ecosystem to drive relevant change to the network infrastructure. This usage of the Terraform provider in conjunction with Consul allows for Consul to act as the core data source to inform updates to the state of the infrastructure.
```
Comment (preetapan) [inline website/pages/docs/nia/index.mdx:6]: Consider expanding the first couple of sentences to bridge the gap between the concept and the tool. Maybe something like
"Services registered in Consul can also be used as the source of truth to feed updates to network infrastructure like load balancers and firewalls.  Network Infrastructure Automation (NIA) relies on a declarative, workflow and service driven network automation architecture. Consul-Terraform-Sync is a new daemon based tool that utlitizes Terraform as.."

File: website/pages/docs/nia/index.mdx [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+---
+
+# Network Infrastructure Automation <sup>Tech Preview</sup>
+
+Network Infrastructure Automation (NIA) enables dynamic updates to network infrastructure devices triggered by service changes. Consul-Terraform-Sync utilizes Consul as a data source that contains networking information about services and monitors those services. Terraform is used as the underlying automation tool and leverages the Terraform provider ecosystem to drive relevant change to the network infrastructure.
+
```
Comment (preetapan) [inline website/pages/docs/nia/index.mdx:12]: drive relevant changes?

File: website/pages/docs/nia/index.mdx [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+Consul-Terraform-Sync executes one or more automation tasks with the most recent service variable values from the Consul service catalog. Each task consists of a runbook automation written as a Consul-Terraform-Sync compatible Terraform module using resources and data sources for the underlying network infrastructure. The `consul-terraform-sync` daemon runs on the same node as a Consul agent.
+
+## Use Cases
+
+**Application teams must wait for manual changes in the network to release, scale up/down and re-deploy their applications.** This creates a bottleneck, especially in frequent workflows related to scaling up/down the application, breaking the DevOps goal of self-service enablement. Consul-Terraform-Sync automates that process which decreases the time in which those changes are made and decreases the possibility of human error.
+
```
Comment (preetapan) [inline website/pages/docs/nia/index.mdx:18]: rewording suggestion "Consul-Terraform-Sync` automates this process, thus decreasing the possibility of human error in manually editing configuration files, as well as decreasing the overall time taken to push out configuration changes.

File: website/pages/docs/nia/index.mdx [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+**Networking and security teams cannot scale processes to the speed and changes needed.** Many network and security teams are struggling to keep up with the rate of change because they are using manual approaches. Even teams which have implemented a degree of network automation are struggling to keep up with the change due to a lack of accurate, real-time source of data to trigger and drive their network automation workflows. Consul-Terraform-Sync runs in near real-time to keep up with the rate of change.
```
Comment (preetapan) [inline website/pages/docs/nia/index.mdx:19]: Would suggest a more direct style of writing here, something like  "Manual approaches don't scale well, causing backlogs in network and security teams. Even in organizations that have some amount of automation (such as scripting), there is a need for an accurate, real-time source of data to trigger and drive their network automation workflows. Consul-Terraform-Sync runs in near real-time to keep up with the rate of change"

File: website/pages/docs/nia/installation/install.mdx (hunk 3 above)
Comment (BeenzSyed) [reply website/pages/docs/nia/installation/install.mdx:11]: 👍 Updated

File: website/pages/docs/nia/index.mdx (hunk 5 above)
Comment (BeenzSyed) [reply website/pages/docs/nia/index.mdx:12]: 👍 Updated

File: website/pages/docs/nia/index.mdx (hunk 6 above)
Comment (BeenzSyed) [reply website/pages/docs/nia/index.mdx:18]: 👍 Updated

File: website/pages/docs/nia/index.mdx (hunk 7 above)
Comment (BeenzSyed) [reply website/pages/docs/nia/index.mdx:19]: 👍 Thank you!

File: website/pages/docs/nia/index.mdx (hunk 4 above)
Comment (BeenzSyed) [reply website/pages/docs/nia/index.mdx]: I updated this but different from the suggestion. It's now - "Network Infrastructure Automation (NIA) is the concept of dynamically updating infrastructure devices triggered by service changes. Consul-Terraform-Sync is a tool that performs NIA and utilizes Consul as a data source that contains networking information about services and monitors those services. Terraform is used as the underlying automation tool and leverages the Terraform provider ecosystem to drive relevant changes to the network infrastructure."

**Pull request merged**